zebra-chain/src/block/header.rs

```diff
@@ -35,10 +34,19 @@ pub struct BlockHeader {
     /// header.
     pub merkle_root_hash: MerkleTreeRootHash,
 
-    /// [Sapling onward] The root LEBS2OSP256(rt) of the Sapling note
+    /// [Pre-Sapling] Reserved. All zeroes.
+    /// [Sapling and Blossom] The root LEBS2OSP256(rt) of the Sapling note
     /// commitment tree corresponding to the final Sapling treestate of
     /// this block.
-    pub final_sapling_root_hash: SaplingNoteTreeRootHash,
+    /// [Heartwood onward] The root of a Merkle Mountain Range tree, which
+    /// commits to various features of the chain's history, including the
+    /// Sapling commitment tree. This commitment supports the FlyClient
+    /// protocol. See ZIP-221 for details.
+    // TODO:
+    //   - replace with an unspecified HistoryRootHash type?
+    // Note that the NetworkUpgrade list is in zebra-consensus, so we can't
+    // parse this field into a HistoryRootHash enum in zebra-chain.
+    pub history_root_hash: [u8; 32],
```

Comment:
What's the relationship between this field and the `HistoryRootHash` enum defined below? I think we might want to change the `BlockHeader` type to be an enum rather than having an enum just for this field, if there are any consensus rules involving the interpretation of the field.

Reply:
This field is the `light client root hash`, an unspecified variant of the HistoryRootHash enum.

Once we know the network upgrade, we can convert the `light client root hash` into the appropriate `HistoryRootHash` enum value.

But at the moment, the network upgrade enum is in zebra-consensus, not zebra-chain. So we can't know the correct HistoryRootHash variant in zebra-chain. I'm not sure I want to shift the network upgrade definitions to zebra-chain.

Once #767 merges, I can make all the necessary changes:
https://github.com/ZcashFoundation/zebra/pull/768#issue-457081416

Reply:
If the network upgrade enum is needed here to parse blocks correctly, we can just move it, I don't think we should leave fields unparsed.

Reply:
See #856, I don't know how to get the BlockHeight and Network while parsing the BlockHeader.